index.d.ts

```diff
@@ -0,0 +1,30 @@
+export type Change<S, M> = [S, Effect<M>?];
+
+export interface Effect<M> {
+  (dispatch: Dispatch<M>): void;
+}
+
+export interface Update<M, S> {
+  (message: M, state: S): Change<S, M>;
+}
+
+export interface View<M, S, V> {
+  (state: S, dispatch: Dispatch<M>): V;
+}
+
+export interface Dispatch<M> {
+  (message: M): void;
+}
+
+export interface Done<S> {
+  (state: S): void;
+}
+
+export interface Program<S, M, V> {
+  readonly init?: Change<S, M>;
+  readonly update: Update<M, S>;
+  readonly view: View<M, S, V>;
```

Comment:
One issue with generic `V` is that for `raj/runtime` `view` is `<State, Message>(state: State, dispatch: Dispatch<Message>): void` as Raj does nothing with the return value. The return value of view is given meaning in the view libraries such as `raj-react`. Typing this we either over generalize the core framework or have multiple Program definitions that end-users would need to reconcile.

I'm not saying this is wrong, just something to consider.

Reply:
Yes, this only makes sense if we consider `raj-react` or similar packages. I've already made typings for `raj-react` so I'll make a similar PR to that repo in order to put the generic `V` in context for discussing these typings.